### PR TITLE
add yum_mysql_community_repo resource

### PR DESCRIPTION
**Pull request by Stromweld**
Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

### Description

adds a resource to create the mysql-community.repo file

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

File: resources/repo.rb [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+         default: false,
+         description: 'Enable or disable mysql tools repo'
+
+property :mysql_cluster_community, [true, false],
+         default: false,
+         description: 'Enable or disable GPG checks mysql cluster community repo'
+
+action :create do
+  description 'Create MySQL Community Repo file'
+  os = case node['platform']
```
Comment (tas50) [inline resources/repo.rb]: You're better off using platform_family here and just doing rhel/amazon

File: resources/repo.rb [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+             else
+               '$releasever'
+             end
+           when 'amazon'
+             case node['platform_version'].to_i
+             when /201./
+               '6'
+             when 2
+               '7'
+             when 3
```
Comment (tas50) [inline resources/repo.rb]: It makes me sad that we have to do this, but there's no way around it

File: resources/repo.rb (hunk 1 above)
Comment (tas50) [inline resources/repo.rb]: you might also just want to extract them into your action class since you already have one.

Comment (Stromweld) [reply resources/repo.rb]: is fedora part of rhel platform_family?

Comment (tas50) [reply resources/repo.rb]: Nope fedora has its own platform family because it's oddly different

Comment (Stromweld) [reply resources/repo.rb]: made suggested change in latest commit

Comment (Stromweld) [reply resources/repo.rb]: made suggested change in latest commit

**Comment by tas50** [comment]
You have a minor cookstyle warning keeping this from going green

**Comment by Stromweld** [comment]
yeah I just caught that.

**Comment by tas50** [comment]
released as 4.1. Thanks

**Pull request merged**